### ISSUE TITLE
Vast.ai provider support

### ISSUE THREAD

**Issue by Nathan-Kr** [apls777/spotty#126]
I used spotty for a while running on AWS instance. Now using it for personnal project, i was looking for a cheaper alternative. 
Browsing reddit i found vast.ai, they describe themself as "the market leader in low-cost cloud GPU rental" wich in fact seems to be true.

Lately i've been working on an integration with spotty on my fork. 
Now that i'm ahead of Poc i was wondering if this feature would be interesting for the project.

If it is, i will rework my code and do the tests as well to submit a PR

![image](https://github.com/spotty-cloud/spotty/assets/66138267/66e5d22b-8ce3-46eb-965f-9386b4d79119)

**Comment by apls777** [comment]
Yes, I know about this provider, it would be great to integrate it with Spotty. Looking forward to your PR!

**Comment by ShivamB25** [comment]
@apls777 are u going to update this ?

**Comment by Nathan-Kr** [comment]
@ShivamB25 didn't submit the PR yet. I forgot this project. I remember it's working on my fork, i can get back to it if it's something spotty users would benefit.
But dont know if project still alive

**Comment by ShivamB25** [comment]
@Nathan-Kr time to rewrite this in rust 😆 the project still works. regardless. i am currently using g5 with aws which isn't cost anything to use (their credits)
run it as a seperate implementation with terraform and in rust

**Comment by ShivamB25** [comment]
@Nathan-Kr is there anyway like discord or anywhere i could contact you?

**Comment by Nathan-Kr** [comment]
@ShivamB25 sure you can add me on discord: kroissan

**Comment by ShivamB25** [comment]
@Nathan-Kr added
